### DISCUSSION TITLE
Project crashes when I click on emitter

### DISCUSSION THREAD

**Discussion by hilarytsai** [mozilla/hubs#3127] [Spoke Questions] [answered]
I have several emitters, one of which I'm trying to delete.  When I click on it or even go to right click and delete, it instantly closes my project.  What might cause this? It's possible it may be very render-heavy with a lot of particles, but I can't even click it to see the properties because the second I do, it crashes.

**Comment by emclaren** [comment]
Hi Hilary! this sounds frustrating. Would you be open to sharing a link to the the spoke project with us (if you prefer to keep it private feel free to email it to hubs@mozilla.com). It's hard to know whats going on without taking a look at the console, which it sounds is going to be hard to open because your project is crashing

**Reply by hilarytsai** [reply]
Sure, @emclaren.  If you scroll down a bit in the list of items and find Emitter 2, that's the one that's crashing.

Rink https://hubs.mozilla.com/scenes/fNLZqid

Thanks!

**Comment by cvanmeurs** [comment, marked as answer]
I can confirm that selecting the emitter crashes Spoke.  I have no idea why this is happening, but I do think I found a solution.

You may want to make a backup of your Spoke scene JUST IN CASE.  (File > Save As...  )

1.  Go into your project, and in the upper-left corner, go to File > export legacy .spoke project
2.  Find where your .spoke file was saved to your computer
3.  Edit the .spoke file with a text editor (I used Notepad on Windows)
4.  Do a search for "particle" and find Particle Emitter 2
5.  Highlight all of the text shown in the attached image (you want to keep the blank space after the comma at the start of the selection, but include the blank space at the end, after the second comma, so that when you delete this block of text, you're left with a comma and a space
6.  Delete this block of text
7.  Save the file
8.  Back in Spoke, choose File > import legacy .spoke project.  You'll get a warning about overwriting your scene.  Click OK.
9.  Choose the .spoke file you just edited and saved on your computer and click Open

That should result in the scene without that emitter present.  Note that index 46 will no longer be referenced.  I'm not sure how this will affect the scene moving forward.  To be safe you could go through and fix all the index numbers, starting with 47, and subtracting 1 until you get to the end of the file, but that seems like a big pain.  I'll ask the programmers if they think this is necessary.

Hope that helps!

![emitter2-delete](https://user-images.githubusercontent.com/63256260/95491051-f572b800-094d-11eb-986d-1f2a3be4ecd0.png)

**Reply by johnshaughnessy** [reply]
@hilarytsai 
I don't know how it got to be this way, but it seems that the `transform` component on that particle emitter node is full of `null`s:
```json
      "name": "Particle Emitter 2",
      "components": [
        {
          "name": "transform",
          "props": {
            "position": {
              "x": null,
              "y": null,
              "z": null
            },
            "rotation": {
              "x": null,
              "y": null,
              "z": 0
            },
            "scale": {
              "x": null,
              "y": null,
              "z": null
            }
          }
        },
```
Changing these values to numbers seems to have fixed the problem for me:
```json
      "name": "Particle Emitter 2",
      "components": [
        {
          "name": "transform",
          "props": {
            "position": {
              "x": 0,
              "y": 0,
              "z": 0
            },
            "rotation": {
              "x": 0,
              "y": 0,
              "z": 0
            },
            "scale": {
              "x": 1,
              "y": 1,
              "z": 1
            }
          }
        },
```

![image](https://user-images.githubusercontent.com/4072106/95496164-45a14880-0955-11eb-9764-9c298045502a.png)

If you have trouble editing the gltf file yourself, I can send you the one I edited (in discord because I can't upload it here). You can then "import legacy .spoke file" from Spoke. 

I'll also file a [bug report](https://github.com/mozilla/Spoke/issues/1040) in Spoke. If you have any idea why the values of the transform became `null`, please add that there.

**Reply by johnshaughnessy** [reply]
@cvanmeurs solution also seems to work and will remove the node entirely. Both of us tested this and it _seems ok_ to just delete the entire node.

**Reply by hilarytsai** [reply]
Thank you so much, @cvanmeurs!!! What a smart solution, I will try this and am sure it'll work.  I really appreciate your help on this issue!! @johnshaughnessy I'm not sure how it got to be that way either.  Interesting find! I will try changing the values. 
 I feel like I had set up the emitter normally, and suddenly it became corrupt for some reason I wasn't aware of.
